### DISCUSSION TITLE
mqtt password including comma character, how?

### DISCUSSION THREAD

**Discussion by oh5lep** [merbanan/rtl_433#2769] [Q&A] [answered]
I have set mqtt password including comma and equal (,=) characters. When I try to run rtl_433 with mqtt output, comma breaks it. 
I have tried double and single quotes, escape character etc, but nothing works. Any help, other than changing the password?

Example password with similar outcome: ddFr3zqpP,bfc/Pr=0
And the result is: Invalid key "bfc/Pr" option.

`rtl_433 -f 433500000 -R 36 -M newmodel -M level -F json -F 'mqtt://localhost:1883,user=mqtt,pass="ddFr3zqpP,bfc/Pr=0",retain=0,devices=rtl_433[/id]'`

**Comment by zuckschwerdt** [comment, marked as answer]
Not possible, sorry. We might add support for Env vars maybe, e.g. `$MQTT_USER` and `$MQTT_PASSWORD`

**Reply by oh5lep** [reply]
Sounds promising.

**Reply by zuckschwerdt** [reply]
Reading MQTT auth from `MQTT_USER` and `MQTT_PASSWORD` env vars added with a863963 now.

**Reply by oh5lep** [reply]
Works, thanks a lot! Only cosmetic thing is that 'examples/rtl_433_mqtt_hass.py' uses `MQTT_USERNAME` for username.

**Reply by zuckschwerdt** [reply]
Good catch! Changed to `MQTT_USERNAME` now.
